src/fuzzers/packet-fuzzer.cc

```diff
@@ -0,0 +1,50 @@
+#include "binpac.h"
+
+#include "Net.h"
+#include "Conn.h"
+#include "Sessions.h"
+#include "analyzer/Analyzer.h"
+#include "analyzer/Manager.h"
+#include "analyzer/protocol/pia/PIA.h"
+#include "analyzer/protocol/tcp/TCP.h"
+
+#include "FuzzBuffer.h"
+#include "fuzzer-setup.h"
+
+#include "pcap/dlt.h"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size)
+	{
+
+	zeek::detail::FuzzBuffer fb{data, size};
+
+	if ( ! fb.Valid() )
+		return 0;
+
+	for ( ; ;  )
+		{
+		auto chunk = fb.Next();
+
+		if ( ! chunk )
+			break;
+
+		Packet pkt;
+		auto timestamp = current_time(true);
+		pkt_timeval ts = {long(timestamp), 0};
```

Comment:
```suggestion
		auto timestamp = 42;
		pkt_timeval ts = {timestamp, 0};
```

Think it will work to use a constant timestamp for everything?  Else I'm wondering if having variable time across inputs or even when trying to reproduce any single input can throw off the fuzzer -- it might see a bunch of timers or other stuff going off as a result of time advancing and think that the current input it used was the interesting cause for that code to execute when it really wasn't.

If we're just trying to evaluate/fuzz the parsing aspects of Zeek, think it shouldn't matter if time doesn't move anywhere.  Maybe there's some other way to deterministically associate time advancements with the fuzzer inputs, though?

Reply:
A pattern i've seen is to just grab the first few bytes from the input if extra data is needed... but I think you are right in that just using 42 shouldn't make a difference.. everything will get flushed/expired at the end anyway.